conans/client/toolchain/__init__.py

```diff
@@ -0,0 +1,35 @@
+# coding=utf-8
+
+import os
+from conans.client.generators.virtualenv import VirtualEnvGenerator
+from conans.client.run_environment import RunEnvironment
+
+
+def write_toolchain(conanfile, path, output):
+    if hasattr(conanfile, "toolchain"):
+        assert callable(conanfile.toolchain), "toolchain should be a callable method"
+        # This is the toolchain
+        tc = conanfile.toolchain()
+        tc.dump(path)
+
+        # Run environment
+        run_environment = RunEnvironment(conanfile).vars
+
+        # Dump also files for the virtualenv
+        venv = VirtualEnvGenerator(conanfile)
```

Comment:
The environment might be better decoupled from the ``write_toolchain()``

Reply:
Sure, if we finally agree that the environment is needed and we use it in this PR I will write its own function. If the environment is going to wait for another iteration, then I should remove these lines and there is nothing to decouple.

Do we add the virtualenv generator by default?

Reply:
Without the environment we cannot adjust the `CMAKE_GENERATOR` (needed to select the VS version). Let's keep this open by now and we'll talk about it.